--- a/arch/mips/econet/init.c	2026-07-02 22:33:08.920471643 +0100
+++ b/arch/mips/econet/init.c	2026-07-02 22:32:28.368629545 +0100
@@ -10,6 +10,7 @@
 #include <linux/irqchip.h>
 
 #include <asm/addrspace.h>
+#include <asm/bootinfo.h>
 #include <asm/io.h>
 #include <asm/time.h>
 #include <asm/prom.h>
 #include <asm/smp-ops.h>
@@ -43,6 +45,7 @@
 void __init plat_mem_setup(void)
 {
 	/* Force baseline 256MB KSEG0 memory pool mapping directly to handle extended boot arguments */
+	add_memory_region(0x80000000, 0x10000000, BOOT_MEM_RAM);
 
 	void *dtb;
 
